Contributing to Ruby on Rails > 2. Reporting an Issue (#72)
campuscode/rails-guides-pt-BR: pt-BR/contributing_to_ruby_on_rails.md

--------------------------------------------------------------------------------

Relatar um Issue
------------------

> O Ruby on Rails usa o [Rastreamento de Issues (Issue Tracking) do GitHub](https://github.com/rails/rails/issues) para rastrear problemas (principalmente erros e contribuições de código novo). e você encontrou um bug no Ruby on Rails, este é o lugar para começar. Você precisará criar uma conta (gratuita) do GitHub para enviar um issue, comentar sobre eles ou criar pull requests.

Author: Assim fica ótimo, @maugravena! Não havia visto essas especificações, realmente é melhor não traduzir algumas palavras e mantê-las em itálico (como bug, por exemplo)!

> NOTA: Os erros na versão mais recente do Ruby on Rails provavelmente receberão mais atenção. Além disso, a equipe principal do Rails está sempre interessada no feedback daqueles que podem dedicar algum tempo para testar o "edge Rails"  (o código para a versão do Rails que está atualmente em desenvolvimento). Mais adiante neste guia, você descobrirá como obter o Edge Rails para testes.

Author: @maugravena, você prefere que eu faça essa alteração?

> NOTA: Os *bugs* na versão mais recente do Ruby on Rails provavelmente receberão mais atenção. Além disso, a equipe principal do Rails está sempre interessada no feedback daqueles que podem dedicar algum tempo para testar o  Rails Instável(ou *Edge Rails*, o código para a versão do Rails que está atualmente em desenvolvimento). Mais adiante neste guia, você descobrirá como obter o Rails Instável para testes.

Reviewer: ```suggestion
NOTE: Os *bugs* na versão mais recente do Ruby on Rails provavelmente receberão mais atenção. Além disso, a equipe principal do Rails está sempre interessada no feedback daqueles que podem dedicar algum tempo para testar o  Rails Instável(ou *Edge Rails*, o código para a versão do Rails que está atualmente em desenvolvimento). Mais adiante neste guia, você descobrirá como obter o Rails Instável para testes.
```
Existem algumas marcações que aparecem como TIP: , NOTE: ou WARNING:. Elas não devem ser traduzidas, pois são interpretadas para renderizar uma caixa de texto de estilo especial na página da aplicação

### Criando um Relatório de *Bug*

Se você encontrou um problema no Ruby on Rails que não é um risco de segurança, faça uma pesquisa no GitHub em [*Issues*](https://github.com/rails/rails/issues), caso já tenha sido relatado. Se você não conseguir encontrar nenhuma *issue* aberta no GitHub que resolva o problema encontrado, sua próxima etapa será [abrir uma nova](https://github.com/rails/rails/issues/new). (Consulte a próxima seção para relatar problemas de segurança.)

> O relatório do problema (*issue report*) deve conter um título e uma descrição clara do problema, no mínimo. Você deve incluir o máximo de informações relevantes possível e, pelo menos, postar um exemplo de código que demonstre o problema. Seria ainda melhor se você pudesse incluir um teste de unidade que mostra como o comportamento esperado não está ocorrendo. Seu objetivo deve ser tornar fácil para você - e outras pessoas - reproduzir o bug e descobrir uma correção.

Reviewer: ```suggestion
O relatório do problema (*issue report*) deve conter, no mínimo, um título e uma descrição clara do problema. Você deve incluir o máximo possível de informações relevantes e, pelo menos, postar um exemplo de código que demonstre o problema. Seria ainda melhor se você pudesse incluir um teste de unidade que mostra como o comportamento esperado não está ocorrendo. Seu objetivo deve ser tornar fácil para você – e outras pessoas – reproduzir o bug e descobrir uma correção.
```

> Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas pulem para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".

Reviewer 2: ```suggestion
Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas se torne automaticamente uma atividade ou que outras pessoas pulem para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".
```
acho que podemos melhorar essa tradução, minha sugestão pode ser apenas um começo para ela.

> Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas pulem para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".

Reviewer 2: ```suggestion
Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas corram para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".
```

> Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas pulem para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".

Reviewer 2: Deixar as palavras não traduzidas em itálico

Author: Vou fazer essas alterações, @maugravena. Obrigada pelas sugestões!

> Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas pulem para corrigi-lo. A criação de um issue como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".

Reviewer 2: ```suggestion
Então, não tenha muitas esperanças! A menos que você tenha um tipo de bug "Código Vermelho, Missão Crítica, o Mundo Está Chegando ao Fim", você está criando este relatório de problemas na esperança de que outras pessoas com o mesmo problema possam colaborar com você para resolvê-lo. Não espere que o relatório de problemas veja automaticamente qualquer atividade ou que outras pessoas pulem para corrigi-lo. A criação de uma *issue* como esse serve principalmente para ajudar você a começar o caminho para corrigir o problema e para que outros o confirmem com um comentário "Estou tendo esse problema também".
```

> ### Criar um caso de teste executável

Reviewer 2: ```suggestion
### Criar um Caso de Teste Executável
```

> 

Reviewer: Alguns trechos estão sem tradução e outros estão sem a referência do inglês. Também seria bom checar se está em ordem

Ter uma maneira de reproduzir seu problema será muito útil para que outras pessoas ajudem a confirmar, investigar e, finalmente, corrigi-lo. Você pode fazer isso fornecendo um caso de teste executável. Para facilitar esse processo, preparamos vários modelos de relatório de *bugs* para você usar como ponto de partida:

> * Modelo de *issue* para registro ativo (modelos, banco de dados): [gem](https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_gem.rb) / [master](https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_master.rb)

Reviewer: Não estamos traduzindo models, controllers, views, Active Record etc.
Estamos subindo em breve um manual de tradução

Você pode compartilhar seu caso de teste executável como um [gist](https://gist.github.com) ou simplesmente colar o conteúdo na descrição do problema.

### Tratamento especial para problemas de segurança

ATENÇÃO: Não relate vulnerabilidades de segurança com relatórios públicos de issues do GitHub.  A [página de política de segurança do Rails](https://rubyonrails.org/security) detalha o procedimento a seguir para problemas de segurança.

### E as Feature Requests?

Por favor, não coloque itens de "feature requests" nos issue do GitHub. Se houver um novo recurso que você deseja ver adicionado ao Ruby on Rails, você precisará escrever o código por conta própria - ou convencer alguém a fazer parceria com você para escrever o código. Mais adiante neste guia, você encontrará instruções detalhadas para propor um patch para o Ruby on Rails. Se você inserir um item da lista de desejos nos Problemas do GitHub sem código, poderá esperar que ele seja marcado como "inválido" assim que for revisado.

> Às vezes, a linha entre 'bug' e 'feature' é difícil de traçar. Geralmente, um recurso é qualquer coisa que adiciona novo comportamento, enquanto um bug é qualquer coisa que causa comportamento incorreto. Às vezes, a equipe principal precisa fazer um julgamento. Dito isto, a distinção geralmente afeta apenas a qual lançamento o seu patch entrará; adoramos envios de recursos! Eles simplesmente não serão transportados para a manutenção de branches.

Reviewer 2: ```suggestion
Às vezes, a linha entre 'bug' e 'feature' é difícil de traçar. Geralmente, um recurso é qualquer coisa que adiciona novo comportamento, enquanto um bug é qualquer coisa que causa comportamento incorreto. Às vezes, a equipe principal precisa fazer um julgamento. Dito isto, a distinção geralmente afeta apenas a qual lançamento o seu patch entrará; adoramos envios de *feature*! Elas simplesmente não serão transportadas para a manutenção de branches.
```

> Se você deseja obter feedback sobre uma ideia para um recurso antes de fazer o trabalho para fazer um patch, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar esse recurso. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. Os issues do GitHub não são um local particularmente bom para as discussões às vezes longas e envolvidas que os feature requests exigem.

Reviewer 2: ```suggestion
Se você deseja obter feedback sobre uma ideia de *feature* antes de começá-la e fazer um *patch*, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar esse recurso. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. Os issues do GitHub não são um local particularmente bom para as discussões às vezes longas e envolvidas que os feature requests exigem.
```

> Se você deseja obter feedback sobre uma ideia para um recurso antes de fazer o trabalho para fazer um patch, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar esse recurso. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. Os issues do GitHub não são um local particularmente bom para as discussões às vezes longas e envolvidas que os feature requests exigem.

Reviewer 2: ```suggestion
Se você deseja obter feedback sobre uma ideia para um recurso antes de fazer o trabalho para fazer um patch, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar essa *feature*. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. Os issues do GitHub não são um local particularmente bom para as discussões às vezes longas e envolvidas que os feature requests exigem.
```

> Se você deseja obter feedback sobre uma ideia para um recurso antes de fazer o trabalho para fazer um patch, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar esse recurso. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. Os issues do GitHub não são um local particularmente bom para as discussões às vezes longas e envolvidas que os feature requests exigem.

Reviewer 2: ```suggestion
Se você deseja obter feedback sobre uma ideia para um recurso antes de fazer o trabalho para fazer um patch, envie um email para a [lista de discussão do rails-core](https://groups.google.com/forum/?fromgroups#!forum/rubyonrails-core). Você pode não receber resposta, o que significa que todos são indiferentes. Você pode encontrar alguém que também esteja interessado em criar esse recurso. Você pode receber um "Isso não será aceito". Mas é o lugar certo para discutir novas idéias. As issues do GitHub não são um local particularmente bom para as discussões, às vezes longas e envolvidas que os *feature requests* exigem.
```
